src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1153,7 +1153,7 @@ private static string GenerateForType(TypeGenerationSpec typeMetadata, string me
 /// </summary>
 public {typeInfoPropertyTypeRef} {typeFriendlyName}
 {{
-    get => _{typeFriendlyName} ??= ({typeInfoPropertyTypeRef}){OptionsInstanceVariableName}.GetTypeInfo(typeof({typeCompilableName}));
```

Comment:
Small question just out of curiosity: with this change the first `GetTypeInfo` method will be (as per #81269):
```csharp
=> Options.GetTypeInfo(type);
```

Which I understand will do a bunch of stuff and eventually resolve the cached type info for the input type from the internal concurrent dictionary from the context (or create one if there isn't a cached version already). I assume there's some variance there in terms of speed compared to the linear check it does today where it just directly loads the JSON type info instance from the generated property?

Is it correct to assume that if the number of types is very small this will be a small regression, but as the number of types grows it should potentially be a bit faster once the cost of looking up from that internal cache dictionary starts taking less than linearly walking over all the generated `if` branches for all possible types? 🤔

Reply:
You're correct in your analysis. This should only have impact on the `JsonSerializerContext` overloads, in that they will now have a performance profile identical to the `JsonSerializerOptions` overloads.